R/render.R

```diff
@@ -891,22 +891,15 @@ render <- function(input,
       # in case the output format turns on the --file-scope flag, run its
       # file_scope function to split the input into multiple files
       input_files <- input
-      if (!is.null(output_format$file_scope) &&
-          length(inputs <- output_format$file_scope(input)) > 1) {
-
-        # add the --file-scope option
-        pandoc_args <- c(pandoc_args, "--file-scope")
-
-        # write the split content into *.split.md files
-        input_files <- unlist(lapply(inputs, function(input) {
-          file <- file_with_meta_ext(input$name, "split", "md")
-          file <- file.path(dirname(input), file)
-          write_utf8(input$content, file)
-          file
-        }))
-
-        # cleanup the split files after render
-        on.exit(unlink(input_files), add = TRUE)
+      if (!is.null(output_format$file_scope) && is.function(output_format$file_scope)) {
```

Comment:
Since `is.function(NULL)` returns `FALSE`, the test `is.null()` should be unnecessary:

```suggestion
      if (is.function(output_format$file_scope)) {
```

Reply:
Oh right! Nicely seen !